src/app.js

```diff
@@ -18,15 +18,39 @@ import Routes from './Routes';
 import config from './config';
 
 // If you want to change the language, Change the imports to the match
-// the wanted locale.
+// the wanted locale. Use the messagesInLocale import to add the translations.
 //
 // Remember to also change the language in the config.js file.
 import localeData from 'react-intl/locale-data/en';
-import messages from './translations/en.json';
+import messagesInLocale from './translations/en.json';
+
 // If you are using a non-english locale with moment library,
 // you should also import time specific formatting rules for that locale
 // e.g. for French: import 'moment/locale/fr';
 
+// Flex template application uses English translations as default.
+// If translation key is missing from e.g. fr.json corresponding key
+// will be added to messages from en.json to prevent missing translation key errors.
+import defaultMessages from './translations/en.json';
+
+let messages;
+
+if (config.locale !== 'en') {
+  const sourceKeys = Object.keys(defaultMessages);
+  const targetKeys = Object.keys(messagesInLocale);
+  const diff = difference(sourceKeys, targetKeys);
+
+  const addedKeys = diff.reduce(function(toAdd, key) {
+    const object = {};
+    object[key] = defaultMessages[key];
+    return Object.assign({}, toAdd, object);
+  }, {});
```

Comment:
Use arrow function instead of anonymous `function` declaration and I think object spread is easier to understand than `Object.assign`.
With those changes this can be turned to oneliner:
```js
const addedKeys = diff.reduce((toAdd, key) => ({ ...toAdd, [key]: defaultMessages[key] }), {});
```

or if you prefer explicit returns:
```js
const addedKeys = diff.reduce((toAdd, key) => {
  return { ...toAdd, [key]: defaultMessages[key] };
}, {});
```

Although, this is probably the easiest to understand:
```js
  const addDefaultTranslation = (translations, missingKey) => ({ ...translations, [missingKey]: defaultMessages[missingKey] });
  const addedKeys = diff.reduce(addDefaultTranslation, messagesInLocale);
```

Reply:
Actually, I'd suggest this:
```js
const addMissingTranslations = (sourceLangTranslations, targetLangTranslations) => {
  const sourceKeys = Object.keys(sourceLangTranslations);
  const targetKeys = Object.keys(targetLangTranslations);
  const missingKeys = difference(sourceKeys, targetKeys);

  const addDefaultTranslation = (translations, missingKey) => 
    ({ ...translations, [missingKey]: sourceLangTranslations[missingKey] });

  return missingKeys.reduce(addDefaultTranslation, targetLangTranslations);
};

const isDefaultLanguageInUse = config.locale === 'en';
const messages = isDefaultLanguageInUse
  ? defaultMessages
  : addMissingTranslations(defaultMessages, messagesInLocale);
```

Reply:
(btw. I didn't test that ^)